biogtr/datasets/data_utils.py

```diff
@@ -0,0 +1,464 @@
+        ]
+    )
+
+    return bbox
+
+
+def pose_bbox(
+    instance: sio.Instance, padding: int, im_shape: ArrayLike
+) -> torch.Tensor:
+    """Calculate bbox around instance pose.
+
+    Args:
+        instance: a labeled instance in a frame,
+        padding: the amount to pad around the pose crop
+        im_shape: the size of the original image in (w,h)
+
+    Returns:
+        Bounding box in [y1, x1, y2, x2] format.
+    """
+    w, h = im_shape
+
+    points = torch.Tensor([[p.x, p.y] for p in instance.points])
+
+    min_x = max(torch.nanmin(points[:, 0]) - padding, 0)
+    min_y = max(torch.nanmin(points[:, 1]) - padding, 0)
+    max_x = min(torch.nanmax(points[:, 0]) + padding, w)
+    max_y = min(torch.nanmax(points[:, 1]) + padding, h)
+
+    bbox = torch.Tensor([min_y, min_x, max_y, max_x])
+    return bbox
+
+
+def resize_and_pad(img: torch.Tensor, output_size: int):
+    """Resize and pad an image to fit a square output size.
+
+    Args:
+        img: Image as a tensor of shape (channels, height, width).
+        output_size: Integer size of height and width of output.
+
+    Returns:
+        The image zero padded to be of shape (channels, output_size, output_size).
+    """
+    # Figure out how to scale without breaking aspect ratio.
+    img_height, img_width = img.shape[-2:]
+    if img_width < img_height:  # taller
+        crop_height = output_size
+        scale = crop_height / img_height
+        crop_width = int(img_width * scale)
+    else:  # wider
+        crop_width = output_size
+        scale = crop_width / img_width
+        crop_height = int(img_height * scale)
+
+    # Scale without breaking aspect ratio.
+    img = tvf.resize(img, size=[crop_height, crop_width])
+
+    # Pad to square.
+    img_height, img_width = img.shape[-2:]
+    hp1 = int((output_size - img_width) / 2)
+    vp1 = int((output_size - img_height) / 2)
+    hp2 = output_size - (img_width + hp1)
+    vp2 = output_size - (img_height + vp1)
+    padding = (hp1, vp1, hp2, vp2)
+    return tvf.pad(img, padding, 0, "constant")
+
+
+def sorted_anchors(labels: sio.Labels) -> list[str]:
+    """Sort anchor names from most instances with that node to least.
+
+    Args:
+        labels: a sleap_io.labels object containing all the labels for that video
+
+    Returns:
+        A list of anchor names sorted by most nodes to least nodes
+    """
+    all_anchors = labels.skeletons[0].node_names
+
+    anchor_counts = {anchor: 0 for anchor in all_anchors}
+
+    for i in range(len(labels)):
+        lf = labels[i]
+        for instance in lf:
+            for anchor in all_anchors:
+                x, y = instance[anchor].x, instance[anchor].y
+                if np.isnan(x) or np.isnan(y):
+                    anchor_counts[anchor] += 1
+
+    sorted_anchors = sorted(anchor_counts.keys(), key=lambda k: anchor_counts[k])
+
+    return sorted_anchors
+
+
+"""
+PARSERS
+"""
+
+
+def parse_trackmate_xml(xml_path: str) -> pd.DataFrame:
```

Comment:
We should consolidate the parsing logic (I can handle this) and add appropriate tests.